cocos/renderer/pipeline/PipelineUBO.cpp

```diff
@@ -236,6 +239,9 @@ void PipelineUBO::updateShadowUBOLightView(const RenderPipeline *pipeline, std::
     auto *     device     = gfx::Device::getInstance();
     auto *       sphere     = sceneData->getSphere();
     auto &     shadowUBO  = bufferView;
+    const auto   isHalfFloatPrecision = device->hasFeature(cc::gfx::Feature::COLOR_HALF_FLOAT) && device->hasFeature(cc::gfx::Feature::TEXTURE_HALF_FLOAT);
```

Comment:
I think can cache the value, it is constant.

Reply:
sure